lib/model/folder_sendrecv.go

```diff
@@ -291,6 +301,15 @@ func (f *sendReceiveFolder) pullerIteration(ignores *ignore.Matcher, ignoresChan
 				if _, ok := f.model.Connection(dev); ok {
 					f.queue.Push(file.Name, file.Size, file.ModTime())
 					changed++
+
+					if changed >= maxQueueLen {
+						// We're done for now; set the finished flag to
+						// false so the caller knows we did not do
+						// everything we could, and stop iterating.
+						finished = false
+						return false
+					}
```

Comment:
Shouldn't this block be outside of the switch? `processDirectly` could get huge too.